ckanext/dataset_subscriptions/actions.py

```diff
@@ -0,0 +1,74 @@
+import ckan.plugins.toolkit as toolkit
+import ckan.logic as logic
+import ckan.lib.email_notifications as email_notifications
+import ckan.model as model
+import ckan.lib.base as base
+from ckan.common import config
+
+
+@toolkit.chained_action
+@toolkit.side_effect_free
+def send_email_notifications(original_action, context, data_dict):
+    email_notifications._notifications_functions = [dms_notification_provider]
+    email_notifications.send_notification = dms_send_notification
+    return original_action(context, data_dict)
+
+def dms_send_notification(user, email_dict):
+    '''Email `email_dict` to `user`.'''
+    import ckan.lib.mailer
+    if not user.get('email'):
+        # FIXME: Raise an exception.
+        return
+    # fix for AWS SES not supporting UTF8 encoding of recepient field
+    # https://docs.aws.amazon.com/cli/latest/reference/ses/send-email.html
+    full_name_latin1 = user['display_name'].encode('latin-1', 'replace')
+    try:
+        ckan.lib.mailer.mail_recipient(full_name_latin1, user['email'],
+                email_dict['subject'], email_dict['body'])
+    except ckan.lib.mailer.MailerException:
+        raise
```

Comment:
this function body looks really ugly. I think you'd get better if you use some quasi decorator here instead:
```
def latin_username_send_notification(user, email_dict):
   user['display_name'] = user['display_name'].encode('latin-1', 'replace')
  return email_notifications.send_notification(user, email_dict)
```
and then above:
```
@toolkit.chained_action
@toolkit.side_effect_free
def send_email_notifications(original_action, context, data_dict):
    email_notifications._notifications_functions = [dms_notification_provider]
    email_notifications.send_notification = latin_username_send_notification
    return original_action(context, data_dict)
```
P.S. I've updated the function name as I think it's better to tell what it is instead of where it belongs.

Reply:
thanks, good idea